src/saveload/afterload.cpp

```diff
@@ -2763,6 +2763,36 @@ bool AfterLoadGame()
 		}
 	}
 
+	if (IsSavegameVersionBefore(SLV_DEPOTID_IN_HANGAR_ORDERS)) {
+		/* Update go to hangar orders so they store the DepotID instead of StationID. */
+		for (Aircraft *a : Aircraft::Iterate()) {
+			if (!a->IsNormalAircraft()) continue;
+			if (a->orders == nullptr) continue;
+			if (a->orders->GetFirstSharedVehicle() != a) continue;
+
+			for (Order *order : a->Orders()) {
+				if (!order->IsType(OT_GOTO_DEPOT)) continue;
+				StationID station_id = order->GetDestination();
+				assert(Station::IsValidID(station_id));
+				Station *st = Station::Get(station_id);
+				assert(st->airport.hangar != nullptr);
+				order->SetDestination(st->airport.hangar->index);
+			}
+		}
+
+		for (Aircraft *a : Aircraft::Iterate()) {
```

Comment:
It would be nice to have a brief comment here explaining why we need to iterate through the aircraft list twice.

Reply:
I will merge the two iterations. One was for the common lists of orders and the other one for the current order of the aircraft.